### PR TITLE
Enable e2e MCP integration with optional presubmit tests

### PR DESCRIPTION
**Pull request by ayj**
This PR enables e2e MCP integration with Pilot and Mixer. It fixes several bugs in Galley and Pilot, adds a new `useMCP` option to the helm charts, and enables MCP in a subset of presubmit tests. The presubmit tests are currently optional. Overtime, additional MCP tests will be turned on and made mandatory as we move towards making MCP the default component configuration mechanism.

Specific changes in this PR:

- add no-op AllowAllChecker to MCP server instead of relying on nil check of an interface value for authz checks - see https://golang.org/doc/faq#nil_error

- fix galley's registration for Istio's RbacConfig CRD

- fix createTime checks in galley's kube converter unit tests

- teach Galley to fix-up complete authentication policy with incomplete mtls policy. Pilot [currently performs this fix](https://github.com/istio/istio/blob/fc8bc2689357e4bada55820e101d17f842f56688/pilot/pkg/networking/plugin/authn/authentication.go#L76-L78) at runtime. This approach does not work if authentication policy is proto encoded as the partially complete oneof field is lost during marshaling. 

- decrease galley's max wait and quiescence duration for publishing downstream configuration pushes. The current e2e and integration tests have random sleep(N) throughout to handle config propagation where N is on the order of seconds.

- add `useMCP` helm option and update pilot and mixer deployments to use MCP instead of k8s for config ingestion when set. This handles both secure and insecure control plane auth cases.

- fix pilot mcp data model integration with ServiceEntry and the service registry controller. Multiple event handlers (per-type) and per-config events are required.
    
- add mcp-e2e-mixer-noauth-v1alpha3-v2 and mcp-e2e-pilot-auth-v1alpha3-v2 circle optional presubmit tests

### CODE REVIEW

File: galley/pkg/kube/converter/converter.go [hunk 1]
```diff
@@ -68,3 +70,54 @@ func legacyMixerResource(_ resource.Info, name string, u *unstructured.Unstructu
...
+	// Alternatively, it might also be possible to customize the marshaling
+	// code to preserve the nil object.
+	for i, peer := range policy.Peers {
+		if mtls, ok := peer.Params.(*authn.PeerAuthenticationMethod_Mtls); ok && mtls.Mtls == nil {
+			policy.Peers[i] = &authn.PeerAuthenticationMethod{
+				&authn.PeerAuthenticationMethod_Mtls{&authn.MutualTls{}},
+			}
+		}
+	}
+
```
Comment (ayj) [inline galley/pkg/kube/converter/converter.go]: cc @ozevren 

The test code below demonstrates the problem. Pilot currently avoids this problem by decoding from the CRD (via JSON/YAML). The partially nil one-of is never proto encoded.

```bash
$ cat auth.go
package main

import (
        "fmt"

        "github.com/gogo/protobuf/proto"
        authn "istio.io/api/authentication/v1alpha1"
)

func main() {
        policy := &authn.Policy{
                Peers: []*authn.PeerAuthenticationMethod{{
                        &authn.PeerAuthenticationMethod_Mtls{},
                }},
        }
        encoded, err := proto.Marshal(policy)
        if err != nil {
                panic(err)
        }

        var decoded authn.Policy
        if err := proto.Unmarshal(encoded, &decoded); err != nil {
                panic(err)
        }

        fmt.Printf("before %+v\n", policy.GetPeers())
        fmt.Printf("decoded %+v\n", decoded.GetPeers())
}

$ go run auth.go
before [mtls:/* nil */ ]
decoded []

```

File: tests/e2e/tests/pilot/authn_policy_test.go [hunk 2]
```diff
@@ -176,7 +182,8 @@ func TestGatewayIngress_AuthN_JWT(t *testing.T) {
 
 	// Configure a route from us.bookinfo.com to "c-v2" only
 	cfgs := &deployableConfig{
-		Namespace: tc.Kube.Namespace,
+		kubeconfig: "/usr/local/google/home/jasonyoung/.kube/config",
```
Comment (ayj) [inline tests/e2e/tests/pilot/authn_policy_test.go]: Temp code that I'll remove before final review and submit

**Comment by ayj** [comment]
cc @ozevren for Mixer and Galley changes
cc @andraxylia, @Nino-K for Pilot changes
cc @geeknoid for circleci changes

File: tests/e2e/framework/kubernetes.go [hunk 3]
```diff
@@ -83,7 +83,7 @@ var (
 	proxyTag           = flag.String("proxy_tag", os.Getenv("TAG"), "Proxy tag")
 	caHub              = flag.String("ca_hub", os.Getenv("HUB"), "Ca hub")
 	caTag              = flag.String("ca_tag", os.Getenv("TAG"), "Ca tag")
-	galleyHub          = flag.String("galley_hub", os.Getenv("HUB"), "Galley hub")
+	galleyHub          = flag.String("galley_hub", os.Getenv("HUB"), "MCP hub")
```
Comment (ozevren) [inline tests/e2e/framework/kubernetes.go]: Should w standardize names here?

**Comment by ayj** [comment]
cc @hklai for circeci changes

File: .circleci/config.yml [hunk 4]
```diff
@@ -176,6 +176,43 @@ jobs:
       - store_test_results:
           path: /go/out/tests
 
+  mcp-e2e-mixer-noauth-v1alpha3-v2:
```
Comment (hklai) [inline .circleci/config.yml]: The "unofficial" naming convention here is e2e-{component}-{features}.

Could you update the new names to follow that too?

Comment (ayj) [reply .circleci/config.yml]: Done

FYI, these separate MCP test variants are temporary until MCP is vetted and becomes the only option (~1.2?)

**Pull request merged**